pkg/collector/corechecks/sbom/processor.go

```diff
@@ -177,11 +177,23 @@ func (p *processor) processHostRefresh() {
 	}
 
 	if err := p.sbomScanner.Scan(scanRequest, p.hostScanOpts, ch); err != nil {
-		log.Errorf("Failed to generate SBOM for host: %s", err)
+		log.Errorf("Failed to trigger SBOM generation for host: %s", err)
+		close(ch)
+		return
 	}
 
 	go func() {
-		result := <-ch
+		result, ok := <-ch
```

Comment:
I don't think having `!ok` can ever happen in this case as we consider SBOM scanner always send something (but never closes the channel).